### PR TITLE
Add a helper for getting Newlines

### PR DESCRIPTION
**Pull request by aozgaa**

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -2535,7 +2528,7 @@ namespace ts {
                 getCancellationToken: () => cancellationToken,
                 getCanonicalFileName,
                 useCaseSensitiveFileNames: () => useCaseSensitivefileNames,
-                getNewLine: () => host.getNewLine ? host.getNewLine() : "\r\n",
+                getNewLine: () => getNewLineOrDefault(host.getNewLine),
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: You actually can't do this because you're losing `getNewLine`'s `this` target.

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -2092,6 +2092,14 @@ namespace ts {
         }
         return carriageReturnLineFeed;
     }
+    
+    /**
+     * The default is CRLF.
+     */   
+    export function getNewLineOrDefault(getNewLine?: () => string): string {
+        return getNewLine ? getNewLine() : carriageReturnLineFeed;
+        var x: number;
```
Comment (DanielRosenwasser) [inline src/compiler/utilities.ts]: Wat

Comment (mhegazy) [inline src/compiler/utilities.ts]: i would move that to services\utilities instead

Comment (aozgaa) [inline src/compiler/utilities.ts]: @DanielRosenwasser Oops.

@mhegazy : okay.

**Pull request merged**